File: frontend/src/shared/types/notification.ts
export type NotificationType =
  | "NEW_ORDER"
  | "NEW_MESSAGE"
  | "PRODUCT_APPROVED"
  | "PRODUCT_REJECTED"
  | "NEW_REVIEW";

export interface NotificationItem {
  id: number;
  user_id: number;
  type: NotificationType;
  payload_json: Record<string, unknown>;
  is_read: boolean;
  created_at: string;
}

File: backend/app/models/notification.py
from __future__ import annotations

import enum
from datetime import UTC, datetime

from sqlalchemy import JSON, Boolean, DateTime, Enum, ForeignKey, Integer
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base


class NotificationType(str, enum.Enum):
    NEW_ORDER = "NEW_ORDER"
    NEW_MESSAGE = "NEW_MESSAGE"
    PRODUCT_APPROVED = "PRODUCT_APPROVED"
    PRODUCT_REJECTED = "PRODUCT_REJECTED"
    NEW_REVIEW = "NEW_REVIEW"


class Notification(Base):
    __tablename__ = "notifications"

    id: Mapped[int] = mapped_column(Integer, primary_key=True)
    user_id: Mapped[int] = mapped_column(Integer, ForeignKey("users.id", ondelete="CASCADE"), index=True)
    type: Mapped[NotificationType] = mapped_column(Enum(NotificationType), index=True)
    payload_json: Mapped[dict] = mapped_column(JSON, default=dict)
    is_read: Mapped[bool] = mapped_column(Boolean, default=False, index=True)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), default=lambda: datetime.now(UTC), index=True)
